controller/src/beerocks/master/db/db.cpp

```diff
@@ -2117,6 +2117,324 @@ bool db::get_hostap_is_acs_enabled(std::string mac)
     return n->hostap->is_acs_enabled;
 }
 
+//
+// Channel Scan
+//
+bool db::set_channel_scan_is_enabled(const std::string &mac, const bool enable)
+{
+    auto n = get_node(mac);
+
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return false;
+    }
+    LOG(DEBUG) << __FUNCTION__ << ", continuous scan is enable = " << int(enable);
+    n->hostap->continuous_scan_config->is_enabled = enable;
+    return true;
+}
+
+bool db::get_channel_scan_is_enabled(const std::string &mac)
+{
+    auto n = get_node(mac);
+
+    if (!n) {
+        LOG(ERROR) << "node not found.... ";
+        return false;
+    } else if (n->get_type() != beerocks::TYPE_SLAVE || n->hostap == nullptr) {
+        LOG(ERROR) << __FUNCTION__ << "node " << mac << " is not a valid hostap!";
+        return false;
+    }
+    return n->hostap->continuous_scan_config->is_enabled;
+}
+
+bool db::set_channel_scan_interval_sec(const std::string &mac, const int interval_sec)
```

Comment:
while it can't be negative, the default is set to -1, in order to force the user to initially set the `interval_sec`, either we declare a default for the `interval_sec` or have the member and function signature not match

Reply:
so just remove the `const`

Reply:
removed all the unneeded `const` from the signatures